workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx

```diff
@@ -52,83 +75,114 @@ const GetAppFlow = () => {
 		},
 	});
 
-	const onCancel = () => {
-		Liferay.Util.navigate(getSiteURL());
-	};
-
-	const onContinue = async (nextStep: StepType) => {
-		setStep(nextStep);
-
-		return;
-	};
-
-	const onPrevious = async (previousStep: StepType) => {
-		setStep(previousStep);
-
-		return;
-	};
-
-	const StepFormComponent: StepComponent = {
-		[StepType.ACCOUNT]: (
-			<AccountSelection
-				onSelectAccount={(account: Account) => {
-					setValue('selectedAccount', account);
-					setShowAccount(true);
-				}}
-			/>
-		),
-	};
-
-	return (
-		<>
-			<ProductCard
-				productId={Number(getUrlParam('productId'))}
-				selectedAccount={getValues('selectedAccount')}
-				setProductToForm={(product: Product) =>
-					setValue('product', product)
-				}
-				showAccount={showAccount}
-			/>
-			<div className="border d-flex flex-column mt-7 p-5 rounded">
-				<div className="d-flex flex-column">
-					<div className="align-self-center h1 mb-6">
-						{sectionProperties[step].title}
-					</div>
-
-					<div>{StepFormComponent[step]}</div>
-				</div>
-				<div className="d-flex justify-content-between mt-5 pt-2">
-					<ClayButton displayType={null} onClick={() => onCancel()}>
-						Cancel
-					</ClayButton>
-					<div className="align-self-end">
-						{sectionProperties[step].backStep !== step && (
-							<ClayButton
-								displayType="secondary"
-								onClick={() =>
-									onPrevious(sectionProperties[step].backStep)
-								}
-							>
-								Back
-							</ClayButton>
-						)}
-						{sectionProperties[step].nextStep && (
-							<ClayButton
-								className="ml-5"
-								onClick={() => {
-									onContinue(
-										sectionProperties[step].nextStep
-									);
-								}}
-							>
-								Continue
-							</ClayButton>
-						)}
-					</div>
-				</div>
-			</div>
-		</>
-	);
+  const { product, selectedAccount } = getValues();
+  const productId = product?.productId;
+
+  const { sku } = useGetProductSkus(product, setEnableTrialMethod);
+  const { channel } = useGetChannelInfo();
+  const { addresses } = useGetAddresses(selectedAccount);
+  const { isFreeApp } = useProductPriceModel(product);
+
+
+  async function handleGetApp() {
+    const productSpecificationValues = await getProductSpecificationValues(
+      productId
+    );
+
+    const orderType = await getProductOrderTypes(productSpecificationValues);
+    const cart = buildNewCart(
+      billingAddress,
+      channel,
+      email,
+      isFreeApp,
+      orderType,
+      product,
+      purchaseOrderNumber,
+      selectedAccount,
+      selectedPaymentMethod,
+      sku
+    );
+
+    const cartResponse = await postCartByPaymentMethod(cart, channel.id);
+
+    await postCheckoutCart({ cartId: cartResponse.id });
+
+    const nextStepsCallbackURL = `${Liferay.ThemeDisplay.getCanonicalURL().replace(
+      "/get-app",
+      ""
+    )}/next-steps?orderId=${encodeURIComponent(cartResponse.id)}`;
```

Comment:
You may want to turn it generic. 
```suggestion
    const nextStepsCallbackURL = (backURL, nextURL) => {`${Liferay.ThemeDisplay.getCanonicalURL().replace(
     `${backURL}`,
      ""
    )}/`${nextURL}`}`;
```

Reply:
Well noticed, thank you!